editor/src/components/shared/scoped-variables.ts

```diff
@@ -83,26 +99,79 @@ function getVariablesFromComponent(
   const jsxComponentVariables = variablesInScopeFromEditorState[elementPathString] ?? {}
   return {
     filePath: jsxComponent?.name ?? 'Component',
-    variables: Object.entries(jsxComponentVariables).flatMap(([name, value]) => {
-      const type = getTypeByValue(value)
-      const variable = {
-        name,
-        value,
-        type,
+    variables: generateVariableTypes(jsxComponentVariables),
+  }
+}
+
+function getComponentPropsInScope(
+  topLevelElements: TopLevelElement[],
+  elementPath: ElementPath,
+  jsxMetadata: ElementInstanceMetadataMap,
+): AllVariablesInScope {
+  const jsxComponent = topLevelElements.find((el) => isUtopiaJSXComponent(el)) as UtopiaJSXComponent
+  const jsxComponentPropNamesUsed = jsxComponent?.propsUsed ?? []
+
+  const jsxComponentProps = omitWithPredicate(
+    getContainerPropsValue(elementPath, jsxMetadata),
+    (key) => !jsxComponentPropNamesUsed.includes(key as string),
+  )
+
+  const name = jsxComponent?.name ?? 'Component'
+  return {
+    filePath: `${name}::props`,
+    variables: generateVariableTypes(jsxComponentProps),
+  }
+}
+
+function getContainerPropsValue(
+  elementPath: ElementPath,
+  jsxMetadata: ElementInstanceMetadataMap,
+): VariableData {
+  const containerPathString = toComponentId(getContainingComponent(elementPath))
+  const containerMetadata = jsxMetadata[containerPathString]?.element.value
```

Comment:
I think it is nicer to use `isLeft`/`isRight`/`foldEither` or other `either` helpers than working with the raw value and checking the type of that with `typeof` later.

Reply:
Thanks!